### ISSUE TITLE
Automatically backport Merged PRs into main to other active branches

### ISSUE THREAD

**Issue by valaparthvi** [rancher/hosted-providers-e2e#205]
Currently, our PRs are merged right into the main, but we do not backport them to other active branches such as release/v2.8, release/v2.9, and release/v2.10.

We could set up a GH automation to open backporting PRs once a PR is merged into `main`.

**Comment by cpinjani** [comment]
I am not sure **all** PRs would be eligible for back-port due shepherd compatibility, different error messages for P1 tests (for eg.  [aks-operator/issues/619](https://github.com/rancher/aks-operator/issues/619) is not in v2.9)
Yes, PRs with CI changes can be.

**Comment by valaparthvi** [comment]
>I am not sure all PRs would be eligible for back-port due shepherd compatibility, different error messages for P1 tests (for eg. https://github.com/rancher/aks-operator/issues/619 is not in v2.9)

You are right, but we could use commands like `/backport 2.10`  in the comments to trigger these actions. And modify the backport tests if needed before merging them.

**Comment by thehejik** [comment]
Good idea, we could definitely try it out to see how it will handle conflicts and so.

But wouldn't be better to just somehow handle shepherd versions in go.mod for different rancher versions instead of having 3 various branches?

**Comment by valaparthvi** [comment]
>But wouldn't be better to just somehow handle shepherd versions in go.mod for different rancher versions instead of having 3 various branches?

I am still not sure I understand what you are trying to say, but my main objective with this PR was to backport PRs related to test case automation, rather than dependency bumps. This backport will be manual, so we can skip the dependency bump PRs.

**Comment by valaparthvi** [comment]
Since we have decided to move away from branching, this issue can be closed.
